### PR TITLE
Release cross builds

### PR DESCRIPTION
**Pull request by ktock**
#253

Release cross builds

This commit adds the following cross builds

- amd64
- arm
- arm64
- ppc64le
- s390x

Note that the test is currently done only against amd64 binaries.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,10 +24,28 @@ jobs:
       run: |
         mkdir ${OUTPUT_DIR}
         RELEASE_TAG="${GITHUB_REF##*/}"
-        TAR_FILE_NAME="stargz-snapshotter-${RELEASE_TAG}-linux-amd64.tar.gz"
+        TAR_FILE_NAME="stargz-snapshotter-${RELEASE_TAG}-linux-${{ matrix.arch }}.tar.gz"
```
Comment (AkihiroSuda) [inline .github/workflows/release.yml:27]: BuildKit and nerdctl uses `-linux-arm-v7.tgz` convention for ARM

File: Dockerfile [hunk 2]
```diff
@@ -41,6 +41,7 @@ RUN apt-get update -y && apt-get install -y libseccomp-dev && \
 
 # Build stargz snapshotter
 FROM golang-base AS snapshotter-dev
+ARG GOARCH
```
Comment (AkihiroSuda) [inline Dockerfile:44]: use TARGETARCH

File: .github/workflows/release.yml (hunk 1 above)
Comment (ktock) [reply .github/workflows/release.yml]: Fixed to use `arm-v7` identifier.

File: Dockerfile (hunk 2 above)
Comment (ktock) [reply Dockerfile]: Fixed this.

**Review by AkihiroSuda — APPROVED** [review]
We could simplify the flow using builds --platform (that's why I suggested using TARGETARCH), but it can be worked out later, so LGTM

**Pull request merged**